### PR TITLE
feat: prevent logind actions for power key and lid

### PR DESCRIPTION
**Pull request by SoumyaRanjanPatnaik**
This commit introduces support for inhibiting logind's default actions for power key press and lid close events. It adds a new module "logind.rs" which contains the necessary code to interact with logind using the zbus library. The main function is updated to setup logind inhibits when the application starts. To support these changes, `zbus` is added as a dependency to Cargo.toml and Cargo.lock. The debian control files are also updated to remove the logind-ignore-lidswitch and logind-ignore-powerkey packages as they are no longer required.

REFERENCE: https://www.freedesktop.org/wiki/Software/systemd/inhibit/

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,3 +11,4 @@ gio = "0.18.4"
 glib = "0.18.5"
 gsettings-macro = "0.1.20"
 swayipc = "3.0.1"
+zbus = "4.2.2"
```
Comment (kgilmer) [inline Cargo.toml:14]: concern: This version will be too recent for `debian-testing` (https://packages.debian.org/testing/rustc)

```
$ volsungr 1.71.1 zbus
Searching zbus versions compatible with rust 1.71.1...
Latest compatible version of zbus = "3.15.2"
```

File: src/logind.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    interface = "org.freedesktop.login1.Manager",
+    default_service = "org.freedesktop.login1",
+    default_path = "/org/freedesktop/login1"
+)]
+pub trait Logind {
+    #[allow(non_snake_case)]
+    fn Inhibit(&self, what: &str, who: &str, why: &str, mode: &str) -> zbus::Result<OwnedFd>;
+}
+
+pub fn setup_logind_inhibits() -> Result<OwnedFd> {
```
Comment (kgilmer) [inline src/logind.rs:15]: so compact!  :smile:

File: src/main.rs [hunk 3]
```diff
@@ -257,9 +259,17 @@ impl PowerSettings {
 }
 fn main() {
     let app = Application::new(Some("org.regolith.powerd"), ApplicationFlags::IS_SERVICE);
-    let hold_guard = ManuallyDrop::new(app.hold());
+
+    // Setup holds for event loop and logind inhibit
+    let hold = (app.hold(), logind::setup_logind_inhibits());
+
+    // Keep the hold guard alive until the end of the program
+    let hold_guard = ManuallyDrop::new(hold);
```
Comment (kgilmer) [inline src/main.rs:267]: nice, TIL

**Review by kgilmer — CHANGES_REQUESTED** [review]
concern about zbus version and Debian's `rustc` version

File: Cargo.toml (hunk 1 above)
Comment (SoumyaRanjanPatnaik) [reply Cargo.toml:14]: Good find. Will change and test with `3.15.2`. Or do we need to go even lower (for supporting older releases)?

Comment (kgilmer) [reply Cargo.toml:14]: I think that's the earliest version, and what we'd decided upon last year as minimum rust version.  Looks like Jammy is on a newer `rustc`: https://packages.ubuntu.com/jammy/rustc

File: src/logind.rs [hunk 4]
```diff
@@ -1,8 +1,8 @@
 use zbus::blocking::Connection;
 use zbus::zvariant::OwnedFd;
-use zbus::{proxy, Result};
+use zbus::{dbus_proxy, Result};
 
-#[proxy(
+#[dbus_proxy(
```
Comment (kgilmer) [inline src/logind.rs:5]: glad to see the change was trivial

**Review by kgilmer — APPROVED** [review]
LGTM!

**Pull request merged**